src/services/functionAppService.ts

```diff
@@ -178,10 +184,18 @@ export class FunctionAppService extends BaseService {
     this.log(`Creating function app: ${this.serviceName}`);
 
     const armService = new ArmService(this.serverless, this.options);
-    let deployment: ArmDeployment = this.config.provider.armTemplate
-      ? await armService.createDeploymentFromConfig(this.config.provider.armTemplate)
-      : await armService.createDeploymentFromType(this.config.provider.type || "consumption");
-
+    const { armTemplate, type } = this.config.provider;
+    let deployment: ArmDeployment = armTemplate
+      ? await armService.createDeploymentFromConfig(armTemplate)
+      : await armService.createDeploymentFromType(type || "consumption");
+
+    if (this.deploymentConfig.runFromBlobUrl) {
+      await this.uploadZippedArtifactToBlobStorage(this.getFunctionZipFile());
```

Comment:
How can you upload the artifact to blob storage before the storage account is provisioned??

Reply:
Fixed.